src/pybind/mgr/cephadm/module.py

```diff
@@ -17,7 +17,7 @@
 import os
 import random
 import tempfile
-import multiprocessing.pool
+from concurrent.futures import ThreadPoolExecutor
```

Comment:
The issue with `multiprocessing.pool` looks quite common: https://bugs.python.org/issue22393. This looks good to me, just one thing to change here in https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/utils.py#L79:
```diff
@@ -76,7 +76,7 @@ def forall_hosts(f: Callable[..., T]) -> Callable[..., List[T]]:
                 raise
 
         assert CephadmOrchestrator.instance is not None
-        return CephadmOrchestrator.instance._worker_pool.map(do_work, vals)
+        return list(CephadmOrchestrator.instance._worker_pool.map(do_work, vals))
 
     return forall_hosts_wrapper
```
What do you think @sebastian-philipp ?